src/Bicep.LangServer.UnitTests/Handlers/BicepDecompileCommandHandlerTests.cs

```diff
@@ -0,0 +1,752 @@
+        public void WithInvalidPath_ShouldThrowArgumentException(string path)
+        {
+            var server = new LanguageServerMock();
+
+            ICompilationManager bicepCompilationManager = Repository.Create<ICompilationManager>().Object;
+            var (handler, _) = CreateCommandHandlers(server);
+
+#pragma warning disable VSTHRD002 // Throws synchronously
+            var sut = () => handler.Handle(new(DocumentUri.File(path)), CancellationToken.None).Result;
+#pragma warning restore VSTHRD002
+
+            sut.Should().Throw<Exception>();
+        }
+
+        [TestMethod]
+        public async Task SimpleJson_ShouldCreateSimpleBicepFile()
+        {
+            string testOutputPath = FileHelper.GetUniqueTestOutputPath(TestContext);
+            string jsonPath = FileHelper.SaveResultFile(TestContext, "main.json", SimpleJson, testOutputPath);
+
+            var server = new LanguageServerMock();
+            server.WindowMock.OnShowDocument();
+
+            var (handler, saveHandler) = CreateCommandHandlers(server);
+            var result = await handler.Handle(
+                new BicepDecompileCommandParams(DocumentUri.File(jsonPath)),
+                CancellationToken.None);
+
+            var expectedBicepPath = Path.ChangeExtension(jsonPath, ".bicep");
+            result.mainBicepPath.Should().BeEquivalentToPath(expectedBicepPath);
+            result.errorMessage.Should().BeNull();
+
+            var saveResult = await saveHandler.Handle(new(result.decompileId, result.outputFiles, true), CancellationToken.None);
+            var output = result.output + saveResult.output;
+
+            output.Should().MatchRegex("Decompiling .*[\\\\/]main\\.json into Bicep\\.\\.\\.");
+            output.Should().MatchRegex("Decompilation complete.");
+            output.Should().MatchRegex("Writing .*[\\\\/]main\\.bicep");
+            File.ReadAllText(expectedBicepPath).Should().BeEquivalentToIgnoringNewlines(SimpleExpectedBicep);
+        }
+
+        [TestMethod]
+        public async Task SimpleJson_ShouldShowDisclaimerMessage()
+        {
+            string testOutputPath = FileHelper.GetUniqueTestOutputPath(TestContext);
+            string jsonPath = FileHelper.SaveResultFile(TestContext, "main.json", SimpleJson, testOutputPath);
+
+            var server = new LanguageServerMock();
+            server.WindowMock.OnShowDocument();
+
+            var (handler, _) = CreateCommandHandlers(server);
+            var result = await handler.Handle(
+                new BicepDecompileCommandParams(DocumentUri.File(jsonPath)),
+                CancellationToken.None);
+
+            result.output.Should().Contain("WARNING: Decompilation is a best-effort process, as there is no guaranteed mapping from ARM JSON to Bicep.");
+            result.output.Should().Contain("You may need to fix warnings and errors in the generated bicep file(s), or decompilation may fail entirely if an accurate conversion is not possible.");
```

Comment:
nit: Can we update it to below?
```
result.output.Should().ContainIgnoringNewlines(TemplateDecompiler.DecompilerDisclaimerMessage);
```

Reply:
I prefer not to do that because if the message gets changed unintentionally in the resx the tests won't catch it.